config/coreboot-p8z77-m_pro.config

```diff
@@ -0,0 +1,19 @@
+CONFIG_USE_BLOBS=y
+CONFIG_VENDOR_ASUS=y
+CONFIG_CBFS_SIZE=0x7E4000
+CONFIG_BOARD_ASUS_P8Z77_M_PRO=y
+CONFIG_HAVE_IFD_BIN=y
+CONFIG_HAVE_ME_BIN=y
+CONFIG_HAVE_GBE_BIN=y
+CONFIG_IFD_BIN_PATH="@BLOB_DIR@/p8z77-m_pro/ifd.bin"
+CONFIG_ME_BIN_PATH="@BLOB_DIR@/p8z77-m_pro/me.bin"
+CONFIG_GBE_BIN_PATH="@BLOB_DIR@/p8z77-m_pro/gbe.bin"
+CONFIG_NO_GFX_INIT=y
+CONFIG_TPM_MEASURED_BOOT=y
+CONFIG_TPM1=y
```

Comment:
Sorry for not having been precise enough.

This coreboot config should also be tpm1  suffixed and board coreboot config file named accordingly so that tpm2 board config will just duplicate and name things correctly.

AFK, will be able to review more correctly in week time to compare Linux config gere with others. This should be considered as a first and quick review.

Reply:
no problem have fixed. 

the kernel config is currently the same as x230 however i think its best to have a separate kernel config for this board as [running swraid](https://github.com/osresearch/heads/pull/1157) is something I've been doing on desktops - which x230s cant (though my w530 and t430 do with the caddy) . 

The raid kernel configs have been unset though for the purposes of this PR, but its going to cause bloat to x230 and others later to have these options enabled. 

`diff linux-p8z77-m_pro.config linux-x230-maximized.config
102,103d101
< # CONFIG_BLK_DEV_MD is not set
< # CONFIG_MD_RAID1 is not set
`
It would be good to figure out a way for making swraid usable that fits in with the your current board and config logical strategy

Reply:
> The raid kernel configs have been unset though for the purposes of this PR, but its going to cause bloat to x230 and others later to have these options enabled.
> 
> diff linux-p8z77-m_pro.config linux-x230-maximized.config 102,103d101 < # CONFIG_BLK_DEV_MD is not set < # CONFIG_MD_RAID1 is not set 
> It would be good to figure out a way for making swraid usable that fits in with the your current board and config logical strategy

I'm confused here thought that was discussed and resolved already. If swraid kernel modules are built as modules, and packed under initrd (modules.cpio) on demand per board config and then loaded early under init, this is no issue.

Reason I readdress this here is that if it is changed under xx30 maximized linux config (and reused for w530/t430, and then ultrabay if anyone cares for x230) the board config would be the only place to switch that config option on, which at build time would pack it, and then be able to use it because exported variable through board config.

Looking for Ethernet driver config option name in board config should lead to working example for that, leading you to Makefile and Linux module config required changes to duplicate this without needing that module to be compiled in kernel as opposed to a kernel module. That kernel module compiled as module would be extensively stripped as opposed to being compiled in, and packed with higher compression rate by xz as a bonus. Everything not required to be compiled in kernel should not be compiled in kernel.

And for the sake of clarity, your linux config for that board could be a symlink to  xx30, and changed later on if something shows as needing be be specific. But the more eyes on the same kernel configs and again less maintainership to tune its size across kernel version bump, which is a really heavy task, which explains its not done often enough and when done, leads to some obscure bugs (as seen for t440p based on 5.x which misbehave as compared to 4.x today booting different distros and not showing early console output through i915/DRM drivers as 4.x counterpart).

Reply:
Other example you might want to look for this is USB keyboard under x230 variant, since USB HID was removed under Linux config as being built in, so that users have to choose a variant if they accept security risk.

This implements all the chain, starting from that board config, the init check, Linux module config change and Makefile changes (if related, I don't remember on top of my head).

You could implement those changes as part of another PR, and have this PR benefit from it simply adding the 
export "SOFT_RAID=y" 
or whatever you see fit.

Reply:
Yeah, just quickly reviewed https://github.com/osresearch/heads/pull/1157

Unfortunately would require a rebase since those changes should be presend under maximized Linux config to deserve a merge (sorry, the need drove merging of different Linux configs otherwise builds failed).

Reply:
Sorry you are right - its been so long i forgot about it.. So basically I could set this in linux-x230-maximized.config and it wont affect other boards, then i just add those modules to my builds, right?

```
CONFIG_MD_RAID1=m
CONFIG_DLK_DEV_MD=m
```

in which case I can just drop the need for the p8z77 kernel config and use x230. I need to push another change anyway as I missed 4096k extra free space in the layout mathematics on the blobs anyway, so ill wait to hear back before pushing that one

Reply:
Yes, that and other changed from older PR.

No need for separate PR, just make that commit real clear about
-adding kernel modules
-adding mdraid module
- makefile requirement
- Linux module mods

And I will accept the change.
Do a reference to past PR in commit and that would be perfect.


Note that calculations for t440p confirmed that math should be ifd max-min in hex= cbfs_size ( where not sure if +- decimal one. Last time I retried that math for closed PR today on x230, I was 1 DEC byte off, soni think it must be +1 on final result, but yet again unsure, can reverify tomorrow).

Reply:
interesting. I had always thought CBFS SIZE was ROM SIZE - (IFD_SIZE+ME_SIZE+GBE_SIZE). Is it really just ROM_SIZE-IFD_SIZE (if so, where does ME and GBE live?)

`FMAP REGION: COREBOOT
Name                           Offset     Type           Size   Comp
cbfs_master_header             0x0        cbfs header        32 none
fallback/romstage              0x80       stage           85168 none
cpu_microcode_blob.bin         0x14dc0    microcode       26624 none
fallback/ramstage              0x1b600    stage          100556 LZMA (223560 decompressed)
config                         0x33f40    raw              1199 none
revision                       0x34440    raw               704 none
build_info                     0x34740    raw                97 none
fallback/dsdt.aml              0x34800    raw              8823 none
vbt.bin                        0x36ac0    raw              1243 LZMA (3902 decompressed)
cmos_layout.bin                0x37000    cmos_layout      1904 none
fallback/postcar               0x377c0    stage           28036 none
fallback/payload               0x3e5c0    simple elf    6975508 none
(empty)                        0x6e5600   null           932836 none
bootblock                      0x7c9200   bootblock       27584 none
`
This is my layout 
`00000000:00000fff fd
0001b000:007fffff bios
00003000:0001afff me
00001000:00002fff gb
`
800000 - 1b000 = 7E5000 (which is what i have set for CONFIG_CBFS_SIZE)



> Do a reference to past PR in commit and that would be perfect.

I recommend we not confuse the two and create dependency. This board being usable is independent of SWRAID and is now using all recommended changes and shares linux-x230-maximized.config so, as this PR stands now we should be able to merge and add this desktop into heads. I can then concentrate on documentation for this board. Once thats done, I will rebase SWRAID under a new PR.